shadow-cljs.edn

```diff
@@ -53,7 +53,12 @@
    :dev {:devtools {:before-load-async status-im.setup.hot-reload/before-reload
                     :after-load-async  status-im.setup.hot-reload/reload
                     :build-notify      status-im.setup.hot-reload/build-notify
-                    :preloads          [re-frisk-remote.preload
+                    :preloads          [;; We require the user namespace (src/user.cljs) so that
+                                        ;; its symbols are available without the dev having to
+                                        ;; evaluate the entire namespace at least once after app
+                                        ;; initialization.
+                                        user
```

Comment:
I'm wondering 🌤️  

What do you think about creating a `dev` namespace with a `dev.user` namespace for local user stuff? I think seeing the `dev` namespace while reading this code would make it a bit clearer that the preload is dev-only. Plus, it would open up the `dev` namespace for other useful dev utilities that others can upload. 

Thoughts?

Reply:
Thanks for the swift review @seanstrom.

I'm kind of confused with the comment. Do you mean like creating a `src/dev/` directory with a `user.cljs` file inside and not the approach suggested in the PR of a git-ignored file in `src/user.cljs`?

I was expecting every dev to know `user` is non-prod code in every major Clojure project and that the preloads are in the `dev` part of the `mobile` target. Do you think that's not clear enough?

The idea of having a `src/dev/` directory I totally like, I just see it as a separate idea from the non-committable `src/user.cljs` file suggested here. They solve different DX problems.

Please, expand my mind @seanstrom :)

Reply:
I think you're right that `src/user.cljs` is common practice for local-user dev stuff, though my mind reads `src/user.cljs` as a namespace for some kind of domain logic for users haha 😸. And that led me to thinking about a combined approach where we create a `dev` namespace but still ignore `src/dev/user.cljs`.

Though I can see that wouldn't result in the cleanest implementation because it would require us to check-in an empty `dev` folder atm.

Reply:
> I think see what you mean @seanstrom. You mean leave `src/dev/user.cljs` as git-ignored, but use that instead of `src/user.cljs` because it's clearer. If that's what you mean I like the suggestion and I can apply it now :)

Oops, I've replied without reading your latest comment here haha. Let me know if you think the empty `dev` folder will be an issue 🙏

Reply:
And I edited this one hehe https://github.com/status-im/status-mobile/pull/19927#discussion_r1592537324

Just to confirm your suggestion before I commit:

- Create a `dev/` directory and add an empty `user` namespace there with brief one-line comment saying it's purpose.
- Update gitignore, shadow-cljs preloads and clj-kondo rule to ignore that file.

What do you think @seanstrom?

Reply:
I'm gonna add a super brief README.md inside `dev/` explaining what it is/could be and also mention about the user namespace. This is a practice we are trying to follow to make file system structures more self-documenting.